preprocessing.py

```diff
@@ -0,0 +1,32 @@
+"""
+Handle preprocessing of the data, such as converting string columns to numeric
+"""
+import pandas as pd
+import numpy as np
+from sklearn.preprocessing import OneHotEncoder, StandardScaler
+
+
+def prepare_data(df):
+    """
+    Splits out the target variable from the rest
+    Converts categorical values to numeric with onehote encoding
+    """
+    df = df.copy()
+    
+    y = df['G3']
+    df.drop('G3', axis=1, inplace=True)
+    
+    cat_cols = df.dtypes[df.dtypes=='object'].index
+    int_cols = df.dtypes[df.dtypes=='int64'].index
+
+    scaler = StandardScaler()
+    X = pd.DataFrame(scaler.fit_transform(df[int_cols]), columns = int_cols)
+    X['intercept'] = 1
+
+    # drop='if_binary' ensures only one dummy column for binary variables
+    transform = OneHotEncoder(drop='if_binary')
```

Comment:
Why are we one-hot encoding instead of dummy encoding? Won't one-hot encoding lead to a non-invertible XTX?

Also, won't one-hot encoding remove the ordering for ordinal variables?

Reply:
We are only one-hot encoding the categorical/boolean variables.

What's the difference between one-hot and dummy encoding?

I checked X.T X and it is invertible